lib/adapter.js

```diff
@@ -1887,28 +1887,40 @@ function Adapter(options) {
                     }
 
                     adapterObjects.getObjectView('system', 'device', {startkey: this.namespace + '.', endkey: this.namespace + '.\u9999', include_docs: true}, (err, _devices) => {
-                        if (_channels && _channels.rows) {
-                            for (let c = _channels.rows.length - 1; c >= 0; c--) {
-                                objects[_channels.rows[c].id] = _channels.rows[c].value;
-                            }
+
+                        if (!adapterObjects) { // if objects is no longer existing, we do not need to unsubscribe
+                            return tools.maybeCallback(callback, objects);
                         }
-                        if (_devices && _devices.rows) {
-                            for (let d = _devices.rows.length - 1; d >= 0; d--) {
-                                objects[_devices.rows[d].id] = _devices.rows[d].value;
+
+                        adapterObjects.getObjectView('system', 'folder', {startkey: this.namespace + '.', endkey: this.namespace + '.\u9999', include_docs: true}, (err, _folders) => {
+                            if (_channels && _channels.rows) {
+                                for (let c = _channels.rows.length - 1; c >= 0; c--) {
+                                    objects[_channels.rows[c].id] = _channels.rows[c].value;
+                                }
                             }
-                        }
-                        if (_states && _states.rows) {
-                            if (options.states) {
-                                this.oStates = {};
+                            if (_devices && _devices.rows) {
+                                for (let d = _devices.rows.length - 1; d >= 0; d--) {
+                                    objects[_devices.rows[d].id] = _devices.rows[d].value;
+                                }
                             }
-                            for (let s = _states.rows.length - 1; s >= 0; s--) {
-                                objects[_states.rows[s].id] = _states.rows[s].value;
-                                if (this.oStates) {
-                                    this.oStates[_states.rows[s].id] = null;
+                            if (_folders && _folders.rows) {
+                                for (let f = _folders.rows.length - 1; f >= 0; f--) {
+                                    objects[_folders.rows[f].id] = _folders.rows[f].value;
                                 }
                             }
-                        }
-                        return tools.maybeCallback(callback, objects);
+                            if (_states && _states.rows) {
+                                if (options.states) {
+                                    this.oStates = {};
```

Comment:
What is this (and the lines below about)? Why are we nulling the states cache when reading all adapter objects?